File: nest/src/core/post/usecase/get-post/impl.ts
import { PostRepository } from "/core/post/domain/repository/post.repository";
import { GetPostUseCase } from "/core/post/usecase/get-post/interface";
import { GetPostInput, GetPostOutput } from "./dto";

export class GetPostUseCaseImpl implements GetPostUseCase {
  constructor(private readonly postRepository: PostRepository) {}
  async execute(getPostInput: GetPostInput): Promise<GetPostOutput> {
    const post = await this.postRepository.getById(getPostInput.id);
    return { post };
  }
}

File: python/src/core/post/usecase/getpost/impl.py
from python.src.core.post.domain.repository.post_repository \
    import PostRepository
from .interface import GetPostUseCase


class GetPostUseCaseImpl(GetPostUseCase):
    def __init__(self, post_repository: PostRepository) -> None:
        super().__init__(post_repository)

    def execute(self):
        return self.post_repository.getPosts()
